foreshadow/serializers.py

```diff
@@ -291,7 +294,13 @@ def dict_deserialize(cls, data):
         else:
             # Cannot use set_params since steps is a required init arg
             # for Pipelines
-            return cls(**params)
+            if issubclass(cls, PipelineSerializerMixin):
+                return cls(**params)
+            else:
```

Comment:
Why was this needed? Is it for ColumnSharer?

Reply:
This is an issue found by @adithyabsk. We were concerned that the `\_\_init\_\_` method of `Pipeline` has a required keyword argument called `steps` and the usual way of "initializing the instance first (assuming the \_\_init\_\_ has no required keyword argument) and then call set_params()" may not work. @adithyabsk please correct me if I misunderstood it.

Reply:
No that makes complete sense and is true. I would add a comment about the required arg for pipelines.

Reply:
will do.

Reply:
Done.

Reply:
@JingJZ160 is right